crates/uv-distribution-types/src/file.rs

```diff
@@ -171,11 +171,20 @@ impl UrlString {
     }
 
     /// Return the [`UrlString`] (as a [`Cow`]) with trailing slash removed.
+    ///
+    /// This matches the semantics of [`Url::pop_if_empty`], which will not trim a trailing slash if
+    /// it's the only path segment, e.g., `https://example.com/` would be unchanged.
     #[must_use]
     pub fn without_trailing_slash(&self) -> Cow<'_, Self> {
         self.as_ref()
             .strip_suffix('/')
-            .map(|path| Cow::Owned(UrlString(SmallString::from(path))))
+            .and_then(|path| {
+                // Only strip the trailing slash if there's _another_ trailing slash that isn't a
+                // part of the scheme.
+                path.rfind('/')
+                    .is_some_and(|r| r > path.find("://").unwrap_or_default() + 2)
```

Comment:
If it doesn't contain `://`, should we return `None` rather than 0? (Again, didn't think about it super hard.)

Reply:
We probably shouldn't add 2, but it seems less surprising to still trim the slash here if there's no scheme

Reply:
I futzed with this a bit and decided to use `split_once` because it's clearer than the find/rfind